sdk/storage/Azure.Storage.DataMovement/MigrationGuide.md

```diff
@@ -146,13 +149,371 @@ TODO
 
 ## Migration Samples
 
-TODO
-- Upload
-- Download
-- S2S
-- Progress reporting
-- Error reporting
-- Pause/Resume
+This section contains side-by-side samples of legacy vs modern library usage of various features.
+These samples are not meant to be exhaustive, but demonstrate a wide variety of uses that may need migration.
+
+- [Upload](#upload)
+  - [Upload single file to blob storage](#upload-single-file-to-blob-storage)
+  - [Upload directory to blob storage](#upload-directory-to-blob-storage)
+- [Download](#download)
+  - [Download single blob](#download-single-blob)
+  - [Download blob directory](#download-blob-directory)
+- [Service to service copy](#copy)
+  - [Copy blob to blob](#copy-blob-to-blob)
+  - [Copy blob to file share](#copy-blob-to-share-file)
+- [Progress reporting](#progress-reporting)
+- [Pause and resume](#pause-and-resume)
+
+### Upload
+
+#### Upload single file to blob storage
+
+**Legacy:**
+```csharp
+// these values provided by your code
+string filePath, containerName, blobName;
+CloudBlobClient client;
+```
+```csharp
+// upload blob
+await TransferManager.UploadAsync(
+    filePath,
+    client.GetContainerReference(containerName).GetBlockBlobReference(blobName));
+```
+**Modern:**
+```csharp
+// these values provided by your code
+string filePath, blobUri;
+BlobsStorageResourceProvider blobs;
+TransferManager transferManager;
+```
+
+```csharp
+// upload blob
+TranferOperation operation = await transferManager.StartTransferAsync(
+    LocalFilesStorageResourceProvider.FromFile(filePath),
+    blobs.FromBlob(blobUri));
```

Comment:
Doesn't have to be here, or maybe in the next commit / PR. But was wondering if we can also add a tidbit that they can also do `.FromClient` if they want to pass the Client over.